### PR TITLE
Delivery Extensions

### PR DESCRIPTION
**Pull request by lmatoti**
Summary of this change:

Delivery Extensions
Deliveries grid menu 'classify_raw_material' to set delivery rmt_code and rmt_classifications

### CODE REVIEW

File: lib/client_rules/client_rmt_rules.rb [hunk 1]
```diff
@@ -218,10 +234,22 @@ def presort_staging_target_location(plant, explain: false)
...
+    def classify_raw_material?(explain: false)
+      return "Allows user to set a delivery's rmt_classifications." if explain
+
+      setting(:classify_raw_material)
+    end
+
+    def show_classify_raw_material_form?(explain: false)
+      return 'Allows user to see the classify_raw_material form.' if explain
+
+      setting(:classify_raw_material) || setting(:classify_raw_material)
```
Comment (jcsjcs) [inline lib/client_rules/client_rmt_rules.rb:252]: These are both the same.

File: lib/masterfiles/repositories/advanced_classifications_repo.rb [hunk 2]
```diff
@@ -122,5 +122,30 @@ def find_rmt_variant_flat(id)
...
+        .where(rmt_classification_type_code: rmt_classification_type_code)
+        .select(:rmt_classification, Sequel[:rmt_classifications][:id])
+        .map(%i[rmt_classification id])
+    end
+
+    def type_code_for_classification(classification_id)
+      DB[:rmt_classifications]
+        .join(:rmt_classification_types, id: :rmt_classification_type_id)
+        .where(Sequel[:rmt_classifications][:id] => classification_id)
+        .select(:rmt_classification_type_code)
```
Comment (jcsjcs) [inline lib/masterfiles/repositories/advanced_classifications_repo.rb:147]: You don't need to specify `select` here, that's handled by the `get`.

File: lib/raw_materials/ui_rules/classify_raw_material_rule.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                             options: @repo.for_select_rmt_classifications_for_type(t),
+                             caption: t,
+                             prompt: true,
+                             required: false }
+      end
+      fields
+    end
+
+    def make_form_object
+      rmt_code_id, rmt_classifications = @repo.select_values(:rmt_deliveries, %i[rmt_code_id rmt_classifications], id: @options[:id])[0]
```
Comment (jcsjcs) [inline lib/raw_materials/ui_rules/classify_raw_material_rule.rb:33]: Use `get` instead of `select_values` here.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

**Comment by lmatoti** [comment]
Delivery Extensions 
Code review changes made

**Pull request merged**